cdf_fabric_replicator/time_series.py

```diff
@@ -75,13 +76,24 @@ def run(self) -> None:
                 time.sleep(sleep_time)
 
     def process_subscriptions(self) -> None:
-        for subscription in self.config.subscriptions:
-            for partition in subscription.partitions:
-                with ThreadPoolExecutor() as executor:
-                    future = executor.submit(
-                        self.process_partition, subscription, partition
-                    )
-                    logging.debug(future.result())
+        if (
+            self.cognite_client.time_series.subscriptions.retrieve(
+                external_id=self.config.subscription.external_id
+            )
+            is None
+        ):
+            try:
+                self.create_subscription(self.config.subscription.num_partitions)
+            except CogniteAPIError as e:
+                logging.error(f"Error creating subscription: {e}")
+                raise e
+
+        for partition in range(0, self.config.subscription.num_partitions):
+            with ThreadPoolExecutor() as executor:
```

Comment:
Why process this in another thread?
Could this lead to a race conditions with the cursor being read and written to the shared state store? 
Do we need to ensure rows in partitions are inserted into the lakehouse in the order they were received from CDF?

Reply:
The original intent was for performance, however, it appears that there could be race conditions with the use of class variables such as update_queue. In addition, if partitions were completed out of order, if could mess up the state store for retry if one of the earlier partitions fails after a later one succeeds and gets marked in the state store. Removing the threading logic until it is identified that more performance is needed at which time it can be re-added in a more thread-safe way.